packages/admin-ui/src/combobox/combobox-multiple.state.ts

```diff
@@ -1,54 +1,80 @@
+import type { Dispatch, ReactNode, SetStateAction } from 'react'
 import { useCallback, useEffect } from 'react'
-import { useCheckboxState } from 'ariakit/checkbox'
 
-import type { ComboboxStateProps } from './combobox.state'
+import type { ComboboxState, ComboboxStateProps } from './combobox.state'
 import { useComboboxState } from './combobox.state'
+import { useCheckboxObjectState } from './selected-multiple.state'
 
-export function useComboboxMultipleState(
-  props: ComboboxMultipleStateProps = {}
-) {
-  const { defaultSelected = [], list, shouldClearOnSelect = true } = props
+export function useComboboxMultipleState<T>(
+  props: ComboboxMultipleStateProps<T> = {}
+): ComboboxMultipleState<T> {
+  const {
+    defaultSelected = [],
+    list,
+    shouldClearOnSelect = true,
+    getOptionValue = (item: T) =>
+      typeof item === 'string' ? item : JSON.stringify(item),
+    renderOption = (item: T) =>
+      typeof item === 'string' ? item : JSON.stringify(item),
+    renderTag: originalRenderTag,
+  } = props
+
+  const renderTag = originalRenderTag || renderOption
 
-  const combobox = useComboboxState({
+  const combobox = useComboboxState<T>({
     // VoiceOver has issues with multi-selectable comboboxes where the DOM focus
     // is on the combobox input, so we set `virtualFocus` to `false` to disable
     // this behavior and put DOM focus on the items.
     virtualFocus: false,
     list,
   })
 
-  const checkbox = useCheckboxState({
+  const checkbox = useCheckboxObjectState({
```

Comment:
```suggestion
  const { select, unselect } = useCheckboxObjectState({
```
then export it directly :)